src/spikeinterface/core/recording_tools.py

```diff
@@ -634,12 +731,42 @@ def get_closest_channels(recording, channel_ids=None, num_channels=None):
     return np.array(closest_channels_inds), np.array(dists)
 
 
+def _noise_level_chunk(segment_index, start_frame, end_frame, worker_ctx):
+    recording = worker_ctx["recording"]
+    
+    one_chunk = recording.get_traces(
+        start_frame=start_frame,
+        end_frame=end_frame,
+        segment_index=segment_index,
+        return_scaled=worker_ctx["return_scaled"],
+    )
+
+
+    if worker_ctx["method"] == "mad":
```

Comment:
Should pre-allocate to reduce memory footprint.

Reply:
we do not have garanty that chaunk have the same size.
I think this will handle more globally when adding `out=...` in `get_traces()`